crates/ddl/src/core/compile/rust/mod.rs

```diff
@@ -142,20 +143,50 @@ fn compile_alias(
             is_copy,
             host_ty,
         } => {
-            // FIXME: Generate unit struct and coercions if necessary!
-
             let doc = core_alias.doc.clone();
             let name = core_alias.name.0.to_pascal_case(); // TODO: name avoidance
-            (
-                core_alias.name.clone(),
-                CompiledItem::Type {
-                    span,
-                    name: name.clone(),
-                    is_copy,
-                    host_ty,
+            let mut derives = Vec::new();
+            if is_copy {
+                derives.push("Copy".to_owned());
+                derives.push("Clone".to_owned());
+            }
+
+            match ty {
+                ty @ rust::Type::If(_, _, _) => match host_ty {
+                    None => unreachable!("type level if for non-format type"),
+                    Some(host_ty) => (
+                        core_alias.name.clone(),
+                        CompiledItem::Type {
+                            span,
+                            name: name.clone(),
+                            is_copy,
+                            host_ty: Some(rust::Type::Var(name.clone())),
+                        },
+                        Some(rust::Item::Struct(rust::StructType {
+                            derives,
+                            doc,
+                            name,
+                            fields: vec![rust::TypeField {
+                                doc: Arc::new([]),
+                                name: "inner".to_owned(),
+                                format_ty: ty,
+                                host_ty,
+                                by_ref: !is_copy,
+                            }],
+                        })),
+                    ),
```

Comment:
This may break once we start composing more complicated expressions together. Ie. we could end up with an if expression in the argument to another type alias once we add functions. I have some ideas how to solve this, which would involve propagating the fact that we've seen an 'arbitrarily complicated' if expression upward (when compiling terms).

This feels a tad ad-hoc though, and might be more reasonably solved if we knew the phase that terms could be evaluated in. Then we'd either be able to partially evaluate (if the dependencies of the term were known at compile time) or convert it to a struct (if we were relying on parse-time information). I'm cool with doing the ad-hoc solution for now though.